options/preprocess.proto

```diff
@@ -7,16 +7,18 @@ import "google/protobuf/descriptor.proto";
 
 // Message level options
 extend google.protobuf.MessageOptions{
-    PreprocessMessageOptions each = 11111;
+    PreprocessMessageOptions message = 11110;
+    PreprocessFieldOptions each = 11111;
 }
 
 message PreprocessMessageOptions {
-    PreprocessString string = 1;
+    string prefunction = 1;
```

Comment:
I think breaking existing interface is not good. PreprocessString is a set of preprocessing actions that is applied to any string in the message.

So should you create:

```
message PreprocessFunction {
    string pre = 1;
    string post = 2;
}

...

PreprocessFunction function (func?) = 2;
```

Reply:
Naming seems to be awkward anyway, why not post_function, so better to pack this into the separate message.

Reply:
Discussed and addressed in commit.